function/fetchFile.js

```diff
@@ -0,0 +1,56 @@
+/* Form of a Google Cloud Function URL (https://cloud.google.com/functions/docs/calling/http)
+   "https://YOUR_REGION-YOUR_PROJECT_ID.cloudfunctions.net/FUNCTION_NAME"
+*/
+
+
+var url = 'https://us-central1-sp19-codeu-35-7727.cloudfunctions.net/predict';
+var data = { } //An array within the displayImage implementation
+
+
+//Just prints the raw JSON data to the console
+fetch(url, {
+  method: 'POST', // or 'PUT'
+  body: data, // data can be `string` or {object}!
+  headers:{
+    'Content-Type': 'application/json'
+  }
+}).then(res => res.json())
+.then(response => console.log(JSON.stringify(response))) //Should return the image's predicted label (aka the breed) and the score
+.catch(error => console.error('Error:', error));
+
+
+
+//Handling the JSON body returned
+var myList = document.querySelector('ul'); //Will contain dog breed on one line and score one the next
+
+fetch(url, {
+  method: 'POST', // or 'PUT'
+  body: data, // data can be `string` or {object}!
+  headers:{
+    'Content-Type': 'application/json'
+  }
+})
+.then(function(response) {
+      if (!response.ok) {
+        throw new Error("HTTP error, status = " + response.status);
+      }
+      return response.json();
+    })
+    .then(function(json) {
+      var breed = document.createElement('li');
+      breed.innerHTML = "Breed: " + response.label;
```

Comment:
Since you're creating a new element each time, submitting a new image will create duplicates.

Suggestion:

* Create HTML that has a `style="display: none;"` attribute wrapping it
* Elements with id for "dog-breed" and "prediction-score" that you can get via `getElementById` on
* When you have a new result, update those only, and set the "style" attribute to `""` (empty string) to show it.

e.g.

```html
<div id="prediction-results" style="display: none">
<span class="label">Breed: </span><span id="dog-breed" class="score"></span>
<span class="label">Confidence: </span><span id="prediction-score" class="score"></span>
</div>
```

```js
# in your function
   let scoreElement = document.getElementById("prediction-score")
   scoreElement.innerText = response.score

   let breedElement = document.getElementById("dog-breed")
   breedElement .innerText = response.label

   let resultDiv = document.getElementById("prediction-results")
   resultDiv.style.display = ""
```

(for example)

Reply:
I understand the HTML items and changes to the function, but I'm a bit confused what you mean by "new result". Each image only gets one response (aka prediction), so how would there be duplicates?

Does that mean that as the function works on the response, there could be a  response.label and response.score that doesn't match what we already saved in document.getElementById("dog-breed") and document.getElementById("prediction-score"); ?